torch/quantization/_quantize_script.py

```diff
@@ -31,40 +31,40 @@ def _prepare_script(model, qconfig_dict, is_dynamic):
     return wrap_cpp_module(torch._C._jit_pass_insert_observers(model._c,
                                                                'forward',
                                                                scripted_qconfig_dict,
-                                                               False,
+                                                               inplace,
                                                                is_dynamic))
 
 def prepare_script(model, qconfig_dict, inplace=False):
-    if not inplace:
-        model = model.copy()
-    return _prepare_script(model, qconfig_dict, is_dynamic=False)
+    assert not inplace, "The inplace support is still in development"
+    return _prepare_script(model, qconfig_dict, inplace, is_dynamic=False)
 
-def prepare_dynamic_script(model, qconfig_dict):
-    return _prepare_script(model, qconfig_dict, is_dynamic=True)
+def prepare_dynamic_script(model, qconfig_dict, inplace=False):
+    assert not inplace, "The inplace support is still in development"
+    return _prepare_script(model, qconfig_dict, inplace, is_dynamic=True)
 
-def _convert_script(model, is_dynamic, debug=False):
+def _convert_script(model, inplace=True, is_dynamic=False, debug=False):
     _check_is_script_module(model)
     model.eval()
-    model = wrap_cpp_module(torch._C._jit_pass_insert_quant_dequant(model._c, 'forward', False, is_dynamic))
+    model = wrap_cpp_module(torch._C._jit_pass_insert_quant_dequant(model._c, 'forward', is_dynamic))
     if not debug:
-        model = wrap_cpp_module(torch._C._jit_pass_quant_finalize(model._c, is_dynamic))
+        model = wrap_cpp_module(torch._C._jit_pass_quant_finalize(model._c, inplace, is_dynamic))
     return model
 
-def convert_script(model, inplace=False, debug=False):
-    if not inplace:
-        model = model.copy()
-    return _convert_script(model, is_dynamic=False, debug=debug)
+def convert_script(model, inplace=True, debug=False):
+    return _convert_script(model, inplace, is_dynamic=False, debug=debug)
 
-def convert_dynamic_script(model, debug=False):
-    return _convert_script(model, is_dynamic=True, debug=debug)
+def convert_dynamic_script(model, inplace=True, debug=False):
+    return _convert_script(model, inplace, is_dynamic=True, debug=debug)
 
-def _quantize_script(model, qconfig_dict, run_fn=None, run_args=None, is_dynamic=False, debug=False):
+def _quantize_script(model, qconfig_dict, run_fn=None, run_args=None, inplace=False, is_dynamic=False, debug=False):
```

Comment:
To be consistent with eager mode, lets make all of the defaults for inplace False,just like it is done here. eager has inplace=False for prepare and convert.

Reply:
it makes sense for `prepare`, but I feel for `convert` we do not really need to do non inplace operation, I was initially thinking about removing the inplace option for `convert`, but later I found that we may have a use case that we prepare once and convert twice(e.g. one with debug, one with no debug) and in that case we'll need the convert to be non inplace.